generic/secrets/security/detected-private-key.yaml

```diff
@@ -1,7 +1,22 @@
 rules:
   - id: detected-private-key
-    pattern-regex: (?i)-{5}begin( [dr]sa| ec| openssh| encrypted)? private key-{5}
-    languages: [regex]
+    patterns:
+    - pattern-either:
+      - patterns:
+        - pattern:
+            -----BEGIN $TYPE PRIVATE KEY-----
+            $KEY
+        - metavariable-regex:
+            metavariable: $TYPE
+            regex: (?i)([dr]sa|ec|openssh|encrypted)?
+      - patterns:
+        - pattern: |
+            -----BEGIN PRIVATE KEY-----
+            $KEY
```

Comment:
I think $KEY will only read up to a slash, since in `generic` mode metavars only capture words.

A few questions:
1. Could we check this at scale? It might not matter too much that $KEY doesn't capture everything.
2. I -think- there's a feature with `pattern-regex` that will map capture groups to metavariables like `$1`, `$2`, etc. If we need to keep the regex but also use entropy detection, we could use that if it works.

Reply:
this rule captured the same results as the previous `private-key` rule, so I think we should be ok with this one.